win32ss/gdi/ntgdi/font.c

```diff
@@ -64,41 +62,39 @@ GreGetKerningPairs(
      return 0;
   }
 
-  pdcattr = dc->pdcattr;
-  TextObj = RealizeFontInit(pdcattr->hlfntNew);
-  DC_UnlockDc(dc);
-
-  if (!TextObj)
+  prfnt = DC_prfnt(dc);
+  if (!prfnt)
   {
      EngSetLastError(ERROR_INVALID_HANDLE);
-     return 0;
+     goto cleanup;
   }
 
-  FontGDI = ObjToGDI(TextObj->Font, FONT);
-  TEXTOBJ_UnlockText(TextObj);
-
-  Count = ftGdiGetKerningPairs(FontGDI,0,NULL);
+  Count = ftGdiGetKerningPairs(prfnt,0,NULL);
+  DC_UnlockDc(dc);
```

Comment:
You're unlocking here, so the `goto cleanup`s below will try to do it again and Do Bad Things.

Reply:
It should assert on that?